feat(teardown): add automated teardown and multi-stack support

liskl opened the pull request:
## Summary
  - Adds automated teardown script with interactive stack detection and fzf selection
  - Creates shared library (lib/common.sh) for common functions
  - Implements UUID-based unique stack identifiers with optional seed phrase
  - Replaces static sleep timers with kubectl wait for reliable dependency checking
  - Enables multiple concurrent stacks with suffix-specific isolated resources
  - Adds prerequisite checking on startup
  - Centralizes all stack files in user home directory (~/.kind-irsa/<suffix>)
  - Cleans up empty directories automatically after teardown
  - Updated cert-manager to v1.18.2

  ## Changes
  - **New**: `full-taredown.sh` - Automated teardown script with interactive selection
  - **New**: `lib/common.sh` - Shared functions for suffix generation, stack listing, resource naming
  - **Enhanced**: `full-setup.sh` - UUID generation, kubectl wait, prerequisite checks, centralized file storage
  - **Updated**: Templates to support centralized stack directory paths
  - **Updated**: Default AWS region changed to us-east-2
  - **Updated**: README with comprehensive teardown instructions

  ## Key Features

  ### Centralized Stack Storage
  - All stack-specific files now stored in `~/.kind-irsa/<suffix>/`
  - Keeps repository root clean and free of generated artifacts
  - Better isolation between multiple deployments
  - Simplified teardown by removing single directory tree
  - Directory structure: keys/, aws/, echoer/ all under stack directory

  ### Automated Teardown
  - Detects deployed stacks by querying AWS S3 for discovery buckets
  - Interactive fzf menu when multiple stacks exist
  - Can specify suffix directly: `./full-taredown.sh <suffix>`
  - Cleans up all AWS resources (IAM roles, OIDC providers, S3 buckets, Kind clusters)
  - Removes entire stack directory (~/.kind-irsa/<suffix>)
  - Automatically removes empty directories

  ### Multi-Stack Support
  - Each stack gets UUID-based unique identifier
  - Optional seed phrase: `./full-setup.sh "my-dev-stack"`
  - All files are isolated in suffix-specific directories
  - Kind clusters named `irsa-<suffix>` for isolation
  - Multiple stacks can run simultaneously

  ### Reliability Improvements
  - Prerequisite checking on startup (kubectl, aws, jq, go, kind, uuidgen, ssh-keygen, openssl)
  - kubectl wait for cert-manager (all 3 deployments) instead of static sleep
  - kubectl wait for pod-identity-webhook instead of static sleep
  - Consistent command-line interface between setup and teardown

  ## Test plan
  - [x] Run `./full-setup.sh` without arguments (generates UUID)
  - [x] Run `./full-setup.sh "my-dev-stack"` with seed phrase
  - [x] Deploy multiple stacks simultaneously with different seeds
  - [x] Run `./full-taredown.sh` with interactive fzf selection
  - [x] Run `./full-taredown.sh <suffix>` with specific suffix
  - [x] Verify all AWS resources are cleaned up (IAM, OIDC, S3)
  - [x] Verify stack directory ~/.kind-irsa/<suffix> is cleaned up
  - [x] Verify repository root remains clean (no keys/, aws/, echoer/ folders)
  - [x] Verify kubectl wait succeeds for cert-manager and webhook

liskl committed:
feat(teardown): add automated teardown and multi-stack support

- Add full-taredown.sh with interactive stack detection and fzf selection
- Create lib/common.sh with shared functions for suffix generation, stack listing, and resource naming
- Implement UUID-based unique stack identifiers (with optional seed phrase argument)
- Add prerequisite checking on startup
- Replace static sleep timers with kubectl wait for cert-manager and pod-identity-webhook
- Make all local files suffix-specific (keys-<suffix>/, aws/*-<suffix>.json, etc.)
- Support multiple concurrent stacks with isolated resources
- Add automatic cleanup of empty directories after teardown
- Update templates to support suffix-specific paths
- Change default AWS region to us-east-2
- Update README with teardown instructions and new prerequisites

liskl commented:
@fanicia, most of this aligns perfectly, but one thing that might be controversial is the updated region from eu-west-1 to us-east-2.

if it's an issue i can revert that one change or make it a configurable option.

liskl committed:
refactor(setup): centralize stack files in user home directory

Move stack-specific files from repository root to ~/.kind-irsa/<suffix>
to keep repository clean and support better isolation between multiple
deployments. All generated files (keys, AWS configs, k8s manifests) are
now stored in dedicated stack directories.

Changes:
- Consolidate keys, aws, and echoer folders into single stack directory
- Update all file paths to use $STACK_DIR variable
- Simplify teardown by removing one directory tree
- Fix template path in kind config (PWD/keys/ instead of PWD/keys-SUFFIX/)

fanicia commented:
Hi @liskl, thanks for the contribution. Looks like some really good improvements. I have been crazy swamped these last few months, so I have not had time to do a proper deep dive into the changes to review properly, yet. I hope to do that soon. Can I get you to make the region configurable again? I am based in EU so I usually default to eu-west-1 🙂

liskl commented:
Totally can do give me a day or two and I'll get that back in :)

liskl committed:
feat(region): make AWS region configurable with eu-west-1 default

This change makes the AWS region fully configurable via the AWS_DEFAULT_REGION
environment variable, addressing PR feedback to restore region configurability.

Changes:
- Updated default region from us-east-2 to eu-west-1 in lib/common.sh
- Fixed Kind cluster template to use HOSTNAME variable instead of hardcoded endpoint
- Fixed s3-echoer job template to use AWS_REGION_PLACEHOLDER
- Updated sed substitution in full-setup.sh to replace region placeholder
- Added comprehensive region configuration documentation to README.md

Users can now override the region: AWS_DEFAULT_REGION=us-west-2 ./full-setup.sh

liskl commented:
@fanicia i just updated the code to default and allow for configuration via ENVAR changes

```
# Use default region (eu-west-1)
./full-setup.sh

# Use a different region
AWS_DEFAULT_REGION=us-west-2 ./full-setup.sh

# Combine with custom stack identifier
AWS_DEFAULT_REGION=ap-southeast-1 ./full-setup.sh "my-dev-stack"
```

liskl commented:
@fanicia any update on this?

fanicia commented:
Sorry for the long wait on this. Had a broken dev environment on my private machine that needed fixing before I could get to this, and very little sparetime these days 😅 I have identified a couple of minor bug fixes that is need fixing, but I'll handle that separately seeing as this is a fork-situation. Thank you for the contribution 🙏
